Apps/WebClient/src/ClientApp/app/views/timeline.vue

```diff
@@ -604,7 +604,11 @@ export default class TimelineComponent extends Vue {
       // Set it to new final page
       this.currentPage = result.length;
     }
-    this.visibleTimelineEntries = result[this.currentPage - 1];
+    if (result.length === 0) {

```

Comment:
I am not 100$ sure on the getPages method. Looks like the filter is being called anytime the current page changes as well. That will cause the logic to re-create the index for the pages without using the other pages in the array (result is an array of arrays).

We should have a separate function that is watching the currentPage and adjust the visible timeline accordingly and use the filter to calculate the max size of the visible page.